realm-annotations-processor/src/main/java/io/realm/processor/RealmJsonTypeHelper.java

```diff
@@ -109,38 +134,56 @@ public static void emitFillRealmListWithJsonValue(String getter, String setter,
                                                       String fieldTypeCanonicalName, String proxyClass,
                                                       JavaWriter writer) throws IOException {
         writer
-            .beginControlFlow("if (!json.isNull(\"%s\"))", fieldName)
-                .emitStatement("obj.%s().clear()", getter)
-                .emitStatement("JSONArray array = json.getJSONArray(\"%s\")", fieldName)
-                .beginControlFlow("for (int i = 0; i < array.length(); i++)")
-                    .emitStatement("%s item = %s.createOrUpdateUsingJsonObject(realm, array.getJSONObject(i), update)",
-                            fieldTypeCanonicalName, proxyClass, fieldTypeCanonicalName)
-                    .emitStatement("obj.%s().add(item)", getter)
+            .beginControlFlow("if (json.has(\"%s\"))", fieldName)
+                .beginControlFlow("if (json.isNull(\"%s\"))", fieldName)
+                    .emitStatement("obj.%s(null)", setter)
+                .nextControlFlow("else")
+                    .emitStatement("obj.%s().clear()", getter)
+                    .emitStatement("JSONArray array = json.getJSONArray(\"%s\")", fieldName)
+                    .beginControlFlow("for (int i = 0; i < array.length(); i++)")
+                        .emitStatement("%s item = %s.createOrUpdateUsingJsonObject(realm, array.getJSONObject(i), update)",
```

Comment:
it seems we're adding RealmObject without checking the null constraints recursively on their respective fields